mintpy/closure_phase_bias.py

```diff
@@ -131,35 +672,156 @@ def main(iargs = None):
+            kernel = isce_utils.gaussian_kernel(5,5,1,1)
+            closurephase_filt = isce_utils.convolve(np.exp(1j*closurephase[i,:,:]), kernel)
+            fid = open(concpdir,mode = 'wb')
+            closurephase_filt.tofile(fid)
+            fid.close()
+            meta['FILE_TYPE']='.int'
+            meta['INTERLEAVE']='BIP'
+            meta['DATA_TYPE']='complex64'
+            meta['BANDS']=1
+            writefile.write_isce_xml(meta, concpdir)
+
+    # compute phase sigma and output
+    for i in range(len(slc_list)-conn):
+        concpname = 'conn'+str(conn)+'_filt_'+'{:03}'.format(i)+'.int'
+        concpdir = os.path.join(cpdir_conn,concpname)
+        concpcorname = 'conn'+str(conn)+'_filt_'+'{:03}'.format(i)+'.cor'
+        concpcordir = os.path.join(cpdir_conn, concpcorname)
+        if not os.path.isfile(concpcordir):
+            isce_utils.estimate_coherence(concpdir, concpcordir)
+
+  #  unwrap
+    for i in range(len(slc_list)-conn):
+        concpname = 'conn'+str(conn)+'_filt_'+'{:03}'.format(i)+'.int'
+        concpdir = os.path.join(cpdir_conn, concpname)
+        concpcorname = 'conn'+str(conn)+'_filt_'+'{:03}'.format(i)+'.cor'
+        concpcordir = os.path.join(cpdir_conn, concpcorname)
+        concpunwname = 'conn'+str(conn)+'_filt_'+'{:03}'.format(i)+'.unw'
+        concpunwdir = os.path.join(cpdir_conn, concpunwname)
+        if not os.path.isfile(concpunwdir):
+            isce_utils.unwrap_snaphu(concpdir,concpcordir,concpunwdir,meta)
+
+  # output accumulated unwrapped closure phase time-series
+    cum_seq_unw_closure_phase(conn,cpdir_conn,length,width,refY,refX, slc_list, meta)
+
+
+def main(iargs = None):
+    inps = cmd_line_parse(iargs)
+    if inps.num_sigma:
+        num_sigma = inps.num_sigma
+    else:
+        num_sigma = 3
+    if inps.action == 'create_mask':
+        creat_cp_mask(inps.ifgram_stack, inps.nl, inps.maxMemory, num_sigma, inps.episilon, inps.outdir)
+
+    if inps.action == 'quick_bias_estimate':
+        maxconn = np.maximum(2,inps.bw) # to make sure we have con-2 closure phase processed
+        if inps.update_cp:
+            for conn in np.arange(2,maxconn+1):
+                compute_unwrap_closure_phase(inps.ifgram_stack, conn, inps.maxMemory, inps.outdir)
+            compute_unwrap_closure_phase(inps.ifgram_stack, inps.nl, inps.maxMemory, inps.outdir)
+        # a quick solution to bias-correction and output diagonal component of Wr (how fast the bias-inducing signal decays with temporal baseline)
+        quick_bias_correction(inps.ifgram_stack, inps.nl, inps.bw, inps.maxMemory, inps.outdir)
+
+    if inps.action == 'bias_estimate':
+        if inps.update_cp:
+            for conn in np.arange(2,inps.bw+2): # to make sure we have con-2 closure phase processed
+                compute_unwrap_closure_phase(inps.ifgram_stack, conn, inps.maxMemory, inps.outdir)
+            compute_unwrap_closure_phase(inps.ifgram_stack, inps.nl, inps.maxMemory, inps.outdir)
```

Comment:
These lines seem to be the same as line 805-809, could we merge them into one, something as below?

```python
if inps.action.endswith('bias_estimate'):
    code block to compute unwrap closure phase
    code block to correct bias
        if inps.action == 'quick_bias_estimate':
            quick_bias_correction()
        else:
            bias_correction()
```

Reply:
Thanks! Modified as suggested.